server/src/main/java/io/pravega/schemaregistry/rules/jsoncompatibility/ArrayTypeComparator.java

```diff
@@ -0,0 +1,133 @@
+package io.pravega.schemaregistry.rules.jsoncompatibility;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Iterators;
+import io.pravega.schemaregistry.rules.CompatibilityChecker;
+
+import java.util.Iterator;
+
+import static io.pravega.schemaregistry.rules.jsoncompatibility.BreakingChangesStore.BreakingChanges;
+
+public class ArrayTypeComparator {
+    JsonCompatibilityChecker jsonCompatibilityChecker;
+    public void setJsonTypeComparator() {
+        this.jsonCompatibilityChecker = new JsonCompatibilityChecker();
+    }
+    public BreakingChanges compareArrays(JsonNode toCheck, JsonNode toCheckAgainst) {
+        if (toCheck.isArray() && toCheckAgainst.isArray())
+            return arraySimpleBodyComparision(toCheck, toCheckAgainst);
+        else {
+            if (checkUniqueItems(toCheck, toCheckAgainst) != null)
+                return checkUniqueItems(toCheck, toCheckAgainst);
+            BreakingChanges minMaxChanges = minMaxItems(toCheck, toCheckAgainst);
+            if (minMaxChanges != null)
+                return minMaxChanges;
+            BreakingChanges additionalItemsChanges = additionalItems(toCheck, toCheckAgainst);
+            if (additionalItemsChanges != null)
+                return additionalItemsChanges;
+            BreakingChanges itemsValidationChanges = itemValidation(toCheck, toCheckAgainst);
+            if(itemsValidationChanges != null)
+                return itemsValidationChanges;
+            //TO DO: Add contains and tupleValidation
```

Comment:
nit: use "todo" instead of "to do" so that the IDE can parse it and highlight it. 
also any TODO in the code should either be accomponied by a github issue OR it should be implemented before the code is merged.

Reply:
Done